crates/matrix-sdk/src/test_utils/mocks/mod.rs

```diff
@@ -2624,6 +2628,13 @@ impl<'a> MockEndpoint<'a, UploadEndpoint> {
     /// Returns a upload endpoint that emulates success, i.e. the media has been
     /// uploaded to the media server and can be accessed using the given
     /// event has been sent with the given [`MxcUri`].
+    /// Expect the filename query param to match the provided value.
+    pub fn expect_filename(self, filename: &str) -> Self {
```

Comment:
I've started to use the `match_` prefix to mean "a parameter must match this value", so lightly suggesting to do it here too. (Strictly speaking it's not "expecting" as it's not asserting; if the filename isn't provided, the mock will silently not catch the request, and it may run into a 404 instead)

…but i realize the equivalent for mime type is called `expect_mime_type`, so it's fun to use `expect_` here, or better to rename the other one to `match_mime_type` as well. Your choice :)